telegram/ext/_applicationbuilder.py

```diff
@@ -0,0 +1,761 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2022
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains the Builder classes for the telegram.ext module."""
+from asyncio import Queue
+from pathlib import Path
+from typing import (
+    TypeVar,
+    Generic,
+    TYPE_CHECKING,
+    Dict,
+    Union,
+    Type,
+    Optional,
+)
+
+from telegram import Bot
+from telegram._utils.types import ODVInput, DVInput, FilePathInput
+from telegram._utils.defaultvalue import DEFAULT_NONE, DefaultValue, DEFAULT_FALSE
+from telegram.ext import Application, JobQueue, ExtBot, ContextTypes, CallbackContext, Updater
+from telegram.request._httpxrequest import HTTPXRequest
+from telegram.ext._utils.types import CCT, UD, CD, BD, BT, JQ
+from telegram.request import BaseRequest
+
+if TYPE_CHECKING:
+    from telegram.ext import (
+        Defaults,
+        BasePersistence,
+    )
+
+# Type hinting is a bit complicated here because we try to get to a sane level of
+# leveraging generics and therefore need a number of type variables.
+InBT = TypeVar('InBT', bound=Bot)
+InJQ = TypeVar('InJQ', bound=Union[None, JobQueue])
+InCCT = TypeVar('InCCT', bound='CallbackContext')
+InUD = TypeVar('InUD')
+InCD = TypeVar('InCD')
+InBD = TypeVar('InBD')
+BuilderType = TypeVar('BuilderType', bound='ApplicationBuilder')
+
+if TYPE_CHECKING:
+    DEF_CCT = CallbackContext.DEFAULT_TYPE  # type: ignore[misc]
+    InitApplicationBuilder = (
+        ApplicationBuilder[  # noqa: F821  # pylint: disable=used-before-assignment
```

Comment:
```suggestion
        'ApplicationBuilder'[  # noqa: F821  # pylint: disable=used-before-assignment
```
maybe make it a string to remove pylint error?

Reply:
doesn't seem to work … you'd have to do `'AppB[…]'` (otherwise you'd be subscripting a string) and that throws a lot of mypy errors …